vortex-ipc/src/messages.rs

```diff
@@ -186,15 +191,63 @@ impl<'a> WriteFlatBuffer for IPCArray<'a> {
             .collect_vec();
         let children = Some(fbb.create_vector(&children));
 
-        fba::Array::create(
+        let stats = collect_array_stats(fbb, self.1);
+
+        vfb::Array::create(
             fbb,
-            &fba::ArrayArgs {
+            &vfb::ArrayArgs {
                 version: Default::default(),
                 has_buffer: column_data.buffer().is_some(),
                 encoding,
                 metadata,
+                stats: Some(stats),
                 children,
             },
         )
     }
 }
+
+/// Computes all stats and uses the results to create an ArrayStats table for the flatbuffer message
+fn collect_array_stats<'a>(
+    fbb: &'_ mut FlatBufferBuilder<'a>,
+    array: &'_ ArrayData,
+) -> WIPOffset<vfb::ArrayStats<'a>> {
+    let trailing_zero_freq = array
+        .statistics()
+        .get_as::<Vec<u64>>(Stat::TrailingZeroFreq)
+        .ok()
+        .map(|v| v.iter().copied().collect_vec())
+        .map(|v| fbb.create_vector(v.as_slice()));
+
+    let bit_width_freq = array
+        .statistics()
+        .get_as::<Vec<u64>>(Stat::BitWidthFreq)
+        .ok()
+        .map(|v| v.iter().copied().collect_vec())
+        .map(|v| fbb.create_vector(v.as_slice()));
+
+    let min = array
+        .statistics()
+        .get(Stat::Min)
+        .map(|min| min.value().write_flatbuffer(fbb));
+
+    let max = array
+        .statistics()
+        .get(Stat::Max)
+        .map(|max| max.value().write_flatbuffer(fbb));
+
+    let stat_args = &vfb::ArrayStatsArgs {
+        min,
+        max,
+        is_sorted: array.statistics().get_as::<bool>(Stat::IsSorted).ok(),
+        is_strict_sorted: array.statistics().get_as::<bool>(Stat::IsStrictSorted).ok(),
+        is_constant: array.statistics().get_as::<bool>(Stat::IsConstant).ok(),
+        run_count: array.statistics().get_as::<u64>(Stat::RunCount).ok(),
```

Comment:
I'm not sure what our guarantee on these stats is... you may want get_as_cast for the counts

Reply:
There is no get_as_cast, but I've added one and switched the counts over 👍

Reply:
We should keep them as constant type, u64 I would say. That was the assumption since the beginning.